### PR TITLE
MS-755 Split session logging and mutex

### PR DESCRIPTION
**Pull request by luhmirin-s**
* Added non-fatals to see logs around session creation and restoration
* Locked every event cache interaction with mutex to prevent any potential race condition

### CODE REVIEW

File: infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt [hunk 1]
```diff
@@ -15,8 +18,11 @@ internal class SessionEventRepositoryImpl @Inject constructor(
     private val sessionDataCache: SessionDataCache,
 ) : SessionEventRepository {
 
-    override suspend fun createSession(): EventScope {
+    private val eventsLock = Mutex()
+
+    override suspend fun createSession(): EventScope = eventsLock.withLock {
         closeAllSessions(EventScopeEndCause.NEW_SESSION)
+        Simber.w("Creating session from DB")
```
Comment (BurningAXE) [inline infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt:25]: I think this will create too much noise

Comment (BurningAXE) [reply infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt:25]: Can we instead log only when we reach the call on line 54:
`?: createSession()`

**Review by BurningAXE — CHANGES_REQUESTED** [review]
Good idea for the locks!
I have a different idea for the new session logging hence the change request.

File: infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt [hunk 2]
```diff
@@ -42,10 +47,19 @@ internal class SessionEventRepositoryImpl @Inject constructor(
...
+        val cachedScope = sessionDataCache.eventScope
+        if (cachedScope != null) return cachedScope
+
+        val restoredScope = localSessionScope()
+        if (restoredScope != null) {
+            Simber.w("Restored session from DB")
+            return restoredScope
+        }
+
+        Simber.w("Creating new session DB")
```
Comment (BurningAXE) [inline infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt:60]: Maybe "Unexpected creation of new session" or something similar :)

**Review by BurningAXE — APPROVED** [review]

File: infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt (hunk 2 above)
Comment (luhmirin-s) [reply infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt:60]: I don't think that it is unexpected in all cases.

**Review by alexandr-simprints — APPROVED** [review]

**Pull request merged**